pruntime/enclave/src/contracts/assets.rs

```diff
@@ -438,15 +587,74 @@ impl contracts::Contract<Command, Request, Response> for Assets {
                     }
                     metadatum.total_supply += amount;
                 } else {
-                    println!("unknown token id: {}", token_id);
+                    println!("unknown token id: {:?}", token_id);
                 }
             } else if let phala::RawEvent::TransferTokenToChain(who, token_id, amount, sequence) = pe {
-                println!("TransferTokenToChain who: {:?}, token id: {:}, amount: {:}", who, token_id, amount);
+                println!("TransferTokenToChain who: {:?}, token id: {:?}, amount: {:}", who, token_id, amount);
                 let transfer_data = TransferTokenData { data: TransferToken { token_id, dest: AccountIdWrapper(who), amount, sequence }, signature: Vec::new() };
                 println!("transfer data:{:?}", transfer_data);
-                self.queue.retain(|x| x.data.sequence > transfer_data.data.sequence);
+                self.queue.retain(|x| {
+                    return match x {
+                        TxQueue::TransferTokenData(tx) => {
+                            if tx.data.sequence > sequence {
+                                return true;
+                            }
+
+                            false
+                        },
+                        _ => false,
+                    }
+                });
+                println!("queue len: {:}", self.queue.len());
+            } else if let phala::RawEvent::TransferXTokenToChain(who, xtoken_id, amount, sequence) = pe {
+                println!("TransferXTokenToChain who: {:?}, xtoken id: {:?}, amount: {:}", who, xtoken_id, amount);
+                self.queue.retain(|x| {
+                    return match x {
+                        TxQueue::TransferXTokenData(tx) => {
+                            if tx.data.sequence > sequence {
+                                return true;
+                            }
+
+                            false
+                        },
+                        _ => false,
+                    }
+                });
                 println!("queue len: {:}", self.queue.len());
             }
```

Comment:
I think we can simplify this a lot. For both events (TranfserToken / TransferXToken), we want to remove the processed events and only keep the unprocessed events (where `tx.data.sequence > sequence`). This logic applies to the both events.

So I'd like to move this condition to the end, and change it to something like:

```rust
            } if ...... {
                .....
            } else {
                let sequence = match pe {
                    phala::RawEvent::TransferTokenToChain(who, token_id, amount, sequence) => {
                        println!("TransferTokenToChain who: {:?}, token id: {:}, amount: {:}", who, token_id, amount);
                        sequence
                    },
                    phala::RawEvent::TransferXTokenToChain(who, xtoken_id, amount, sequence) => {
                        println!("TransferXTokenToChain who: {:?}, xtoken id: {:?}, amount: {:}", who, xtoken_id, amount);
                        sequence
                    },
                    _ => return
                }
                self.queue.retain(|x| {
                    let sequence_in_queue = match x {
                        TxQueue::TransferTokenData(tx) => tx.data.sequence,
                        TxQueue:: TransferXTokenData(tx) => tx.data.sequence,
                    };
                    sequence_in_queue > sequence
                };
            }
```